syntax = "proto3";

package fixbug; // declare the namespace

// indicate to generate service code and rpc method description
option cc_generic_services = true;

message ResultCode {
    int32 errcode = 1;
    bytes errmsg = 2;
}

// class for login, name, pwd
message LoginRequest {
    bytes name = 1;
    bytes pwd = 2;
}

// class for login response
message LoginResponse {
    ResultCode result = 1;
    bool success = 3;
}

message GetFriendListsRequest {
    uint32 userid = 1;
}

message User {
    bytes name = 1;
    uint32 age = 2;
    enum Sex{
        MAN= 0;
        WOMAN = 1;
    }
    Sex sex = 3;
}

message GetFriendListsResponse {
    ResultCode result = 1;
    repeated User friend_list = 2; // list of friends
}

service UserServiceRpc {
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc GetFriendLists(GetFriendListsRequest) returns(GetFriendListsResponse);
}
